js/ui/endSessionDialog.js

```diff
@@ -769,6 +843,21 @@ var EndSessionDialog = new Lang.Class({
                 this._type = DialogType.UPGRADE_RESTART;
         }
 
+        this._syncUpdateStatus();
+
+        /* If this._parameters has already saved parameters, it means that we have shown
+         * warning dialog(UPDATES_SCHEDULED DialogType) to the user. So, now skip it and
+         * show regular shutdown dialog.
+         */
```

Comment:
Either rework this comment to be more descriptive and shorter, or remove it whatsoever. It doesn't really add to the codebase.